File: source/fortran/structure_variables.f90
module structure_variables
  !! author: J. Morris (UKAEA)
  !!
  !! Module containing global variables relating to the support structure
  !!
  !!### References
  !!
  !! -
#ifndef dp
  use, intrinsic :: iso_fortran_env, only: dp=>real64
#endif

  implicit none

  public

  real(dp) :: aintmass
  !! intercoil structure mass (kg)

  real(dp) :: clgsmass
  !! gravity support structure for TF coil, PF coil and intercoil support systems (kg)

  real(dp) :: coldmass
  !! total mass of components at cryogenic temperatures (kg)

  real(dp) :: fncmass
  !! PF coil outer support fence mass (kg)

  real(dp) :: gsmass
  !! reactor core gravity support mass (kg)
end module structure_variables

File: process/data_structure/structure_variables.py
from dataclasses import dataclass


@dataclass
class StructureData:
    """Data describing the structure."""

    aintmass: float = 0.0
    """intercoil structure mass (kg)"""

    clgsmass: float = 0.0
    """gravity support structure for TF coil, PF coil and intercoil support systems (kg)"""

    coldmass: float = 0.0
    """total mass of components at cryogenic temperatures (kg)"""

    fncmass: float = 0.0
    """PF coil outer support fence mass (kg)"""

    gsmass: float = 0.0
    """reactor core gravity support mass (kg)"""


CREATE_DICTS_FROM_DATACLASS = StructureData
